### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by lambdalisue**
- Install Neovim (HEAD) as a release build
- Enable themis.vim and use neovim instead in themis (while Vim in travis is too old)
- Use pytest instead
- Use vint instead
- Add `make install`
- Add `make lint`
- Simplify travis.yml
- Simplify Makefile
- Fix existing issues (vint/flake8/mypy/...)

### CODE REVIEW

**Comment by lambdalisue** [comment]
Done. Please review this @Shougo

File: .travis.yml [hunk 1]
```diff
@@ -8,13 +8,11 @@ python:
...
 install:
-    - pip install neovim flake8 nose mock pytest mypy-lang
-
-before_script:
-    - git clone https://github.com/syngan/vim-vimlint /tmp/vim-vimlint
-    - git clone https://github.com/ynkdir/vim-vimlparser /tmp/vim-vimlparser
-    - git clone https://github.com/thinca/vim-themis
+    # Install Neovim via Gist
+    # https://gist.github.com/lambdalisue/0f461f6a91743895394e63a9a1c65232
+    - curl -sL git.io/v18DU | bash
```
Comment (Shougo) [inline .travis.yml]: Why it uses the shortcut instead of the full URI?

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
-	nosetests -v rplugin/python3
-	flake8 rplugin/ --exclude rplugin/python3/denite/prompt/docs/conf.py
+PATH := ./vim-themis/bin:$(PATH)
+export THEMIS_VIM  := nvim
+export THEMIS_ARGS := -e -s --headless
+export THEMIS_HOME := ./vim-themis
+
+
+install: vim-themis
+
```
Comment (Shougo) [inline Makefile]: It is not needed line.

**Comment by Shougo** [comment]
Almost good.

File: .travis.yml (hunk 1 above)
Comment (lambdalisue) [inline .travis.yml]: My personal preference. I'll use a full URL if you want.

Comment (Shougo) [inline .travis.yml]: OK.  Please change it.

**Comment by Shougo** [comment]
Merged.  Thank you!

**Pull request merged**